From: Sergio Costas <rastersoft@gmail.com>
Date: Fri, 29 Apr 2022 20:13:53 +0200
Subject: Fix 'GDK_IS_MONITOR (monitor)' failed message

When a computer has no monitors connected, DING fills the log
with the error:

DING: (gjs:10210): Gdk-CRITICAL **: 09:58:45.854: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed

This patch fixes it.

Fix https://gitlab.com/rastersoft/desktop-icons-ng/-/issues/229

Bug-Ubuntu: https://pad.lv/1978331
---
 extension.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extension.js b/extension.js
index bd5a90e..976077e 100644
--- a/extension.js
+++ b/extension.js
@@ -273,7 +273,7 @@ function disable() {
 }
 
 function updateDesktopGeometry() {
-    if (data.actionGroup) {
+    if (data.actionGroup && (Main.layoutManager.monitors.length != 0)) {
         data.actionGroup.change_action_state('desktopGeometry', getDesktopGeometry());
     }
 }
